core/resource/service.go

```diff
@@ -150,17 +150,24 @@ func (s *Service) sync(ctx context.Context, r Resource) (*Resource, error) {
 	m, err := s.moduleRegistry.Get(r.Kind)
 	if err != nil {
 		r.Status = StatusError
-		return &r, err
+	} else {
+		r.Status, err = m.Apply(r)
```

Comment:
i am actually bit confused as to what is the expected behaviour here to be honest. need to discuss it. @rohilsurana @ishanarya0

Reply:
For now i have ignored the error returned by Apply() and left a TODO in place for us to resolve later.